### PR TITLE
Move container configuration types to api/types/container.

### PR DESCRIPTION
**Pull request by calavera**
That way the structs can be shared between the API client library, the
docker client and the docker daemon without importing the whole docker
project as a library.

All the validations remain in the runconfig package. It's up to client and server to check values.

/cc @dnephin 

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by tiborvass** [comment]
@calavera needs rebase :P

**Comment by calavera** [comment]
I've rebased these changes and split them in smaller chunks. The last one is still slightly big, but hopefully won't be too hard to review.

File: api/types/container/host_config.go.back [hunk 1]
```diff
@@ -0,0 +1,224 @@
+package container
```
Comment (jfrazelle) [inline api/types/container/host_config.go.back]: why .back?

File: vendor/src/github.com/docker/go-connections/MAINTAINERS [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Waterfront maintainers file
```
Comment (jfrazelle) [inline vendor/src/github.com/docker/go-connections/MAINTAINERS]: lol waterfront hahaha cute :)

File: api/types/container/host_config.go.back (hunk 1 above)
Comment (calavera) [inline api/types/container/host_config.go.back]: bad commit, removed!

File: vendor/src/github.com/docker/go-connections/MAINTAINERS (hunk 2 above)
Comment (calavera) [inline vendor/src/github.com/docker/go-connections/MAINTAINERS]: hahaha, copy/paste sucks.

**Comment by thaJeztah** [comment]
One nit about tagging go-connections. Looked through the changes, and didn't see strange things :)

**Comment by jfrazelle** [comment]
LGTM

**Comment by calavera** [comment]
@thaJeztah there is a v0.1.0 tag, there are no code changes between the tag and this commit, only docs changed between them.

**Comment by thaJeztah** [comment]
@calavera should probably be used then :innocent: (in a follow up?)

**Comment by calavera** [comment]
@thaJeztah the tag is older than the commit :smile: I'll be happy to tag a v0.2.0 and follow up as soon as we have more code in there.

**Comment by thaJeztah** [comment]
@calavera hehe, no worries, it's just a nit, so "lgtm"

**Comment by icecrime** [comment]
Ping @duglin @MHBauer!

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**